packages/runtime-core/src/components/BaseTransition.ts

```diff
@@ -465,7 +466,8 @@ function getKeepAliveChild(vnode: VNode): VNode | undefined {
       __DEV__ && vnode.component
       ? vnode.component.subTree
       : vnode.children
-        ? ((vnode.children as VNodeArrayChildren)[0] as VNode)
+        ? ((vnode.children as VNodeArrayChildren)[0] as VNode) ||
```

Comment:
Should use `vnode.shapeFlag` to check for `ShapeFlags.ARRAY_CHILDREN` and `ShapeFlags.SLOTS_CHILDREN`

Reply:
I'm not sure the new commit addresses all the usage scenarios